meson.build

```diff
@@ -19,66 +18,97 @@ compiler_args = [
   '-Wno-microsoft-exception-spec',
 ]
 
-link_args = [
-  '-static',
-  '-static-libgcc',
-  '-static-libstdc++',
-  # We need to set the section alignment for debug symbols to
-  # work properly as well as avoiding a memcpy from the Wine loader.
-  '-Wl,--file-alignment=4096',
-]
-
-# Wine's built-in back traces only work with dwarf2 symbols
-if get_option('debug') and target_machine.system() == 'windows'
-  compiler_args += [
-    '-gstrict-dwarf',
-    '-gdwarf-2',
-  ]
-endif
+link_args = []
 
 if get_option('build_id')
   link_args += [
     '-Wl,--build-id',
   ]
 endif
 
-if cpu_family == 'x86'
+if platform == 'windows'
+  compiler_args += [
+    '-DNOMINMAX',
+    '-D_WIN32_WINNT=0xa00',
+  ]
+
   link_args += [
-    '-Wl,--enable-stdcall-fixup',
-    '-Wl,--add-stdcall-alias',
+    '-static',
+    '-static-libgcc',
+    '-static-libstdc++',
+    # We need to set the section alignment for debug symbols to
+    # work properly as well as avoiding a memcpy from the Wine loader.
+    '-Wl,--file-alignment=4096',
+  ]
+
+  # Wine's built-in back traces only work with dwarf2 symbols
+  if get_option('debug')
+    compiler_args += [
+      '-gstrict-dwarf',
+      '-gdwarf-2',
+    ]
+  endif
+
+  # Enable stdcall fixup on 32-bit
+  if cpu_family == 'x86'
+    link_args += [
+      '-Wl,--enable-stdcall-fixup',
+      '-Wl,--add-stdcall-alias',
   ]
+  endif
+
+  if (cpu_family == 'x86_64')
+    dxvk_library_path = meson.source_root() + '/lib'
+  else
+    dxvk_library_path = meson.source_root() + '/lib32'
+  endif
+
+  lib_vulkan  = cpp.find_library('vulkan-1', dirs : dxvk_library_path)
+  lib_d3d9    = cpp.find_library('d3d9')
+  lib_d3d11   = cpp.find_library('d3d11')
+  lib_dxgi    = cpp.find_library('dxgi')
+  lib_d3dcompiler_43 = cpp.find_library('d3dcompiler_43', dirs : dxvk_library_path)
```

Comment:
`meson.source_root()` is deprecated because it cannot sensibly be used while still supporting usage as subprojects.

On the other hand, I think what you wanted was `meson.current_source_dir()` anyway, and that only coincidentally happened to be the same.

Starting with Meson 0.56, which is too high for your minimum requirement, you can use `meson.project_source_root()` instead, but that's primarily useful when you want to get the location from inside a subdir.

Reply:
This is only moving some.code around but ye we can fix it